Sources/Datadog/Kronos/KronosClock.swift

```diff
@@ -125,7 +125,11 @@ internal final class KronosClock: KronosClockProtocol {
     ) {
         self.loadFromDefaults()
 
-        KronosNTPClient().query(pool: pool, numberOfSamples: samples) { offset, done, total in
+        KronosNTPClient().query(pool: pool, numberOfSamples: samples) { [weak self] offset, done, total in
+            guard let self = self else {
+                return
+            }
```

Comment:
Prevent from invoking callbacks if `self` is released.

Reply:
Interestingly, I'm somehow sure it will fix [this unobvious flakiness](https://app.datadoghq.com/ci/test/AQAAAYHXYfyfwZBI4wAAAABBWUhYWWZ5ZkFBQ2lRZkZmYktVNExqaUo?colorBy=service&env=ci&index=citest&spanID=12466522618453451672&spanViewType=metadata).